lib/members/member.ex

```diff
@@ -0,0 +1,40 @@
+defmodule GlassFactoryApi.Members.Member do
+  @moduledoc """
+  Defines the Member of organization.
+  """
+
+  alias GlassFactoryApi.Members.Member
+
+  @type t() :: %Member{
+          name: String.t(),
+          email: String.t(),
+          id: integer(),
+          archived: boolean(),
+          capacity: non_neg_integer(),
+          freelancer: boolean(),
+          joined_at: String.t()
+        }
+
+  defstruct [:name, :email, :id, :archived, :capacity, :freelancer, :joined_at]
+
+  @doc """
+  Parse a map into a Member struct
+
+  ## Examples
+
+      iex> GlassFactoryApi.Member.to_struct(%name: "John Doe", id: 100, email: "john.doe@example.org")
+      %Member{
+        name: "John Doe",
+        id: 100,
+        email: "john.doe@example.org",
+        archived: false,
+        capacity: 6.0,
+        freelancer: false,
+        joined_at: "2019-01-01"
+      }
+  """
+  def to_struct(member) do
```

Comment:
Maybe here we can use `Kernel.struct/2` instead of converting by hand.
e.g. `def to_struct(member_attrs), do: struct(__MODULE__, member_attrs)`

Reply:
I'm using `struct/2` [here](https://github.com/Cendrao/glass_factory_api/pull/3/files#diff-3c9166a5096f0430c7ed4d35df06726eR38), but as API results came in a string format list, I've to symbolize(or atomize? ) it first.

Reply:
Oops, I see, my bad, missed that one 😅

I'm really not that into for calling `String.to_atom` on unknown strings that may come in the response, you can get [some drawbacks](https://engineering.klarna.com/monitoring-erlang-atoms-c1d6a741328e) since there's a limit on how many atoms you can store in your system. I see a couple of workarounds on this:

- One of them is to [use `String.to_existing_atom(key)`](https://til.hashrocket.com/posts/gkwwfy9xvw-converting-strings-to-atoms-safely), so it will only convert for atoms that already exist in the system, for example, the ones you defined on Member's `destruct`

- Another one is to convert the keys by hand, something along the lines of pattern matching the keys and creating structs ourselves
  e.g. `def to_struct(%{"name" => name, ...}), do: %__MODULE__{name: name}`

- If changing the library that we are using for JSON encoding/decoding is an option, we could also make `Poison` handle this for us, since it has an option to [automagically convert maps into structs](https://elixirforum.com/t/how-to-decode-a-json-into-a-struct-safely/14331/3)
  e.g. `body |> Poision.decode(as: %Member{})

I'm leaning more on the side of letting `Poison` handle this for us, WDYT?

Reply:
Using `Poison`'s automagically convert might not take us to the same memory drawback? 🤔 

I'm thinking in convert the keys by hand:

```elixir
%Member{
name: attrs["name"],
id: attrs["id"]
...
```

Reply:
I think you are right, it makes more sense. 😄